### ISSUE TITLE
feat: YaRN long-context beyond native 262144 for Qwen3.6-14B-A3B

### ISSUE THREAD

**Issue by JLFN** [JLFN/mistral.rs#1]
Feature request: YaRN long-context serving beyond the native 262144 tokens
for Qwen3.6-14B-A3B FableVibes Q4_K_M on the mistral.rs deployment.

Background

The YaRN implementation already exists and is mechanism-verified. It was
built on the branch feat/qwen35-yarn-long-context (archived and deleted
2026-08-16; this issue is the record of the plan). What exists:

- CLI flags --rope-scaling yarn, --rope-scale, --yarn-orig-ctx and
  --override-ctx, with RopeOverride plumbed through GGUF synthesis.
- A llama.cpp-exact new_partial_yarn rotary, golden-tested at position
  262145.
- Deployment values for a 320000-token target: scale 1.220703125
  (320000/262144), orig 262144.
- The hybrid paged cache (10 of 40 layers hold paged KV) and the 4-bit F4
  KV cache (~5 KB/token, packed U8 with per-token scales) were built in
  the same unit so the KV pool fits the 12 GB card at 320000 tokens
  (~1.6 GB).

The native 262144 path (no YaRN) is the current production deployment and
is unaffected by this request.

What needs to be done

1. Full 320000-token serve verification of the YaRN path. Launch with
   --rope-scaling yarn --rope-scale 1.220703125 --yarn-orig-ctx 262144
   --pa-context-len 320000 --pa-cache-type f4, run the incremental
   context proof to 320000 with a recall probe of early content, and
   confirm VRAM stays within the 12 GB card across incremental turns.
2. Seeded F4-vs-BF16 logit comparison at scale. The F4 commit
   (54a89061c) lists this as an outstanding known limitation: verify the
   dequantized F4 attention logits match the BF16 reference within
   tolerance on the yarn path.
3. F4 cache sizing fix (cosmetic). The paged-attention sizing math counts
   the U8 storage as 1 byte per element; real packed F4 is 0.5 bytes per
   element. The startup log over-reports the pool 2x ("Allocating 2560 MB"
   vs ~1280 MB real at 262144) and --pa-memory-* budgets under-allocate
   2x. The change is a one-liner at
   mistralrs-core/src/paged_attention/mod.rs (F4 dtype size = 0.5).
   Proposed once before and reverted by the user; re-propose only after
   the items above are settled.
4. A verified launcher variant for the over-native context, analogous to
   /data/run-qwen_mistral.sh, plus the matching open-grok model config
   if the over-native context is wanted in the UI.
5. Optional: upstream the YaRN support to EricLBuehler/mistral.rs with
   tests (golden rotary vectors at positions past the native context,
   GGUF rope-override round-trip, and a serve smoke test).

Acceptance criteria

- A request past 262144 tokens (target 320000) prefills and decodes with
  correct recall of early content.
- VRAM stays within the 12 GB card across incremental turns with no OOM
  and no ratchet.
- The startup log reports the real cache footprint (~1.6 GB at 320000,
  not the 2x figure).
- Golden rotary vectors match llama.cpp at positions beyond 262144.

Notes

- The implementation branch feat/qwen35-yarn-long-context was deleted on
  2026-08-16 at the user's request; its commits remain reachable from
  feat/qwen35-native-context-fix, whose base is 54a89061c (the F4
  commit). The YaRN rotary code lives in mistralrs-core.
- Port 8891 is exclusive with llama.cpp (/data/run-qwen.sh); any
  over-native test run must follow the same GPU/port discipline as the
  native launcher.
